[DOC] Add details for setting Kafka broker env vars (#1884)
strimzi/strimzi-kafka-operator: documentation/book/con-customizing-containers.adoc

// This assembly is included in the following assemblies:
//
// assembly-customizing-deployments.adoc

[id='con-customizing-containers-{context}']
= Customizing containers with environment variables

You can set custom environment variables for Kafka broker containers by using the `kafkaContainer` template property in `Kafka.spec.kafka`.
These environment variables are defined under the `env` field as a list of objects with `name` and `value` fields.

> The following example shows two custom environment variables set for a Kafka broker:

Reviewer: Insert a new paragraph here.

// This assembly is included in the following assemblies:
//
// assembly-customizing-deployments.adoc

[id='con-customizing-containers-{context}']
= Customizing Containers

> Using the `kafkaContainer` template field you can set environment variables to be applied to the Kafka broker containers.

Reviewer 2: ```suggestion
Using the `kafkaContainer` template property, you can set environment variables for Kafka broker containers.
```

> Environment variables should be supplied under the `env` field as a list of objects with `name` and `value` fields. 

Reviewer 2: ```suggestion
Environment variables are supplied under the `env` field as a list of objects with `name` and `value` fields. 
```

The following example shows two custom environment variables being set for the Kafka broker:

[source,yaml,subs=attributes+]
----
# ...
spec:
    kafka:
        template:
            kafkaContainer:
                env:
                    - name: TEST_ENV_1
                      value: test.env.one
                    - name: TEST_ENV_2
                      value: test.env.two
# ...
----

> Environment variables used internally by Strimzi typically (although not always) begin with the `KAFKA_` prefix and should be avoided. 

Reviewer: Do not mention the exceptions unless you define them here or provide a link to documentation where they are covered.

> If an environment variable is set that is already in use by Strimzi then it will be ignored and a warning will be posted in the log output.

Reviewer 2: ```suggestion
If an environment variable is set that is already in use by Strimzi, it is ignored. A warning is posted in the log output.
```
